src/client/app/villains/villains/villains.component.ts

```diff
@@ -22,35 +21,39 @@ export class VillainSearchComponent implements OnDestroy, OnInit {
   selectedVillain: Villain = null;
 
   filteredVillains$: Observable<Villain[]>;
-  villains$: Observable<Villain[]>;
   loading$: Observable<boolean>;
   dataSource$ = this.appSelectors.dataSource$();
 
   private onDestroy = new Subject();
   private villainDispatcher: EntityDispatcher<Villain>;
   private villainSelector: EntitySelectors$<Villain>;
+  private villainAction$: EntityActions<EntityAction<Villain>>;
 
   constructor(
-    private entityService: EntityService,
+    entityService: EntityService,
     private appSelectors: AppSelectors,
     private toast: ToastService
   ) {
     this.villainDispatcher = entityService.getDispatcher(Villain);
     this.villainSelector = entityService.getSelectors$(Villain);
+    this.villainAction$ = entityService.getEntityActions$(Villain);
+
   }
 
   ngOnInit() {
     this.filteredVillains$ = this.villainSelector.selectFilteredEntities$;
-    this.villains$ = this.villainSelector.selectAll$;
     this.loading$ = this.villainSelector.selectLoading$;
 
     this.dataSource$
       .pipe(takeUntil(this.onDestroy), distinctUntilChanged())
       .subscribe((value: string) => this.getVillains());
 
-    this.villains$
-      .pipe(takeUntil(this.onDestroy), skip(1))
-      .subscribe(villains => this.toast.openSnackBar('Fetched Villains', 'GET'));
+    this.villainAction$
+      .filter(ea => ea.op.includes('_SUCCESS') || ea.op.includes('_ERROR'))
```

Comment:
i like this, but we should make it less string sensitive.

Reply:
Done.